pdf/pdfdocument.cpp

```diff
@@ -162,6 +217,16 @@ void PDFDocument::requestPage(int index, int size, QQuickWindow *window, QRect s
     d->thread->queueJob(job);
 }
 
+void PDFDocument::requestArea(int index, int size, QQuickWindow *window, QRect subpart)
+{
+    if (!isLoaded() || isLocked() || subpart.isEmpty())
+        return;
+
+    RenderPageJob* job = new RenderPageJob(index, size, window, subpart,
+                                           RenderPageJob::RenderPageType::Area);
```

Comment:
I did it like that to put more sense into the API, even if a bit more clumpsy. In fact the rendering job is already unified with an enum to decide between Page and Area. The best would have been to expose this enum in pdfcanvas which does the request and pdfjob that stores it, but pdfjob is not visible from pdfcanvas which only address pdfdocument.

What you suggest would have the following API (correct me if I misinterpret your suggestion):
pdfcanvas.cpp:
enum {Page, Area};
m_document->requestPage(…, Page) or m_document->requestPage(…, Area)

pdfdocument.h:
requestPage(…, int flag);

pdfjob.h:
RenderPageJob(…, int flag);

IMHO, when reading this API, the pro is that its simple, to request a page, call requestPage() and details are implementation specific hidden in flag argument. The con is that flag has no mean when not explained from pdfcanvas. What do you prefer ?